### PR TITLE
Tail block caching

### PR DESCRIPTION
**Pull request by mmstick**
- Caches the tail block in `Path::new(config::CACHE).join("tail")`
- Checks the modified time of the file and renews it if it is older than 1 day
- {Ser,Deser}ializes with bincode + serde
- Fixes the delay seen in https://github.com/pop-os/firmware-manager/issues/60 on a high-latency connection each time the view is refreshed

Requires https://github.com/pop-os/buildchain/pull/11
Fixes https://github.com/pop-os/firmware-manager/issues/60

### CODE REVIEW

**Review by jackpot51 — DISMISSED** [review]

File: src/lib.rs
Comment (shirshak55) [inline src/lib.rs:297]: unix line ending?

**Review by bflanagin — COMMENTED** [review]
Tests run to verify the fix for pop-os/firmware-manager#60

1. Got baseline latency info using speedtest.net 
* ping: 15ms
2. Changed wifi router to known dirty channel and proceeded outside. 
* ping: 220ms (its a really dirty channel)

Verified a delay using the stock firmware manager interface on refresh, though it was still rather minor on my connection.

Went through the same procedure with the fix applied. Small delay on first view, but subsequent refreshes didn't exhibit the delay. Older version had a consistent small delay. 

Regressions:
* None found
* Test Run: Uninstalled and reinstalled firmware on logitech receiver via interface. Unplugged / plugged in receiver. 

Notes not associated with this PR but notable: On unplug the firmware management interface, in Gnome Control Center, refreshes and removes the device automatically. On plugging in the device the interface does not refresh automatically, though there is a small visual disturbance which could be it attempting to refresh.  Will test with the current version to see if this bug has been fixed elsewhere.

I think its ready to approve @leviport unless you can think of another way / anything else to test.

**Comment by leviport** [comment]
I'd like if we made sure that laptop firmware updates still works as expected with this installed. The lemp9 should prompt you for an update if you have the currently proposed firmware update installed: https://github.com/system76/firmware/pull/279 . This is more of a regression test though, so you probably don't need to stand in the yard to induce latency while testing.

You should probably only test that after the conflicts on this branch are resolved, but if it looks good after that, then I'd say it's ready for approval.

**Comment by bflanagin** [comment]
Completely dropped the ball on this. The firmware interface worked as expected. If the conflicts aren't an issue then we can approve.

**Comment by jackpot51** [comment]
@mmstick will need to rebase this

**Comment by mmstick** [comment]
Rebased it

**Review by jackpot51 — APPROVED** [review]

**Review by bflanagin — APPROVED** [review]
Re ran tests from the last go. No regressions detected.

**Pull request merged**
